src/redisjson.rs

```diff
@@ -503,4 +505,42 @@ pub mod type_methods {
             }
         }
     }
+
+    pub unsafe extern "C" fn aof_rewrite(
+        aof: *mut raw::RedisModuleIO,
+        key: *mut raw::RedisModuleString,
+        value: *mut c_void,
+    ) {
+        let json = &*(value as *const RedisJSON);
+        let doc_str = json.data.to_string();
+        let key = RedisString::from_ptr(key).unwrap(); // FIXME: Why does it crash when I use key directly and pass "s" to the fmt string??
+        eprintln!("In AOF rewrite for {}", key);
+        if let Some(value_index) = &json.value_index {
+            let index_name = &value_index.index_name;
+            raw::RedisModule_EmitAOF.unwrap()(
+                aof,
+                b"JSON.SET\0".as_ptr() as *const c_char,
+                b"bcbcb\0".as_ptr() as *const c_char,
+                key.as_ptr(),
+                key.len(),
+                b"$\0",
+                doc_str.as_ptr(),
+                doc_str.len(),
+                b"INDEX\0",
+                index_name.as_ptr(),
+                index_name.len(),
+            );
+        } else {
+            raw::RedisModule_EmitAOF.unwrap()(
+                aof,
+                b"JSON.SET\0".as_ptr() as *const c_char,
+                b"bcb\0".as_ptr() as *const c_char,
+                key.as_ptr(),
+                key.len(),
+                b"$\0".as_ptr(),
```

Comment:
Why is the `.as_ptr()` needed, when above it's just `b"$\0"`?

Reply:
no reason, will fix.